app/hooks/use-nostr-profile.ts

```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from "react"
+import * as Keychain from "react-native-keychain"
+import { nip19, generateSecretKey } from "nostr-tools"
+
+const useNostrProfile = () => {
+  const KEYCHAIN_NOSTRCREDS_KEY = "nostr_creds_key"
+  const [nostrSecretKey, setNostrSecretKey] = useState<string>("")
+
+  useEffect(() => {
+    const initializeNostrProfile = async () => {
+      try {
+        console.log("Looking for nostr creds in keychain")
+        const credentials = await Keychain.getInternetCredentials(KEYCHAIN_NOSTRCREDS_KEY)
+        if (credentials) {
+          console.log("nostr creds found in keychain", credentials)
+          setNostrSecretKey(credentials.password)
+          return
+        }
+        const nostrSecret = nip19.nsecEncode(generateSecretKey())
+        console.log("Generated new nostr secret:", nostrSecret)
```

Comment:
need to remove console.logs before merging to main

Reply:
done